infinitered/ExpensifyApp
# Adds Client-side Violations when creating money requests

### Details


### Fixed Issues

$ https://github.com/Expensify/App/issues/31092
PROPOSAL: 


### Tests


#### General Tests


#### Test Violations added to Transaction

**To Test:**
- Ensure that `canUseViolations` in Permissions returns true
- Follow the instructions [here](https://expensify.slack.com/archives/C05EQBQG6E5/p1695182901255769), then create an report that violates one of those policies.

**Test transactionViolations_ are added clientSide to ONYXDB**

- [ ] After correctly setting up the policy and violations requirements create a new money request within that policy
- [ ] Navigate to the `ONYXDB` -> `keyvaluepairs` within your browser storage tab
- [ ] Identify the transaction that was created with the dollar amount you specified, should look like `transactions_someID`
- [ ] Verify a transaction violations object with the corresponding ID like `transactionViolations_SameIDasTransaction`

### Offline tests


Same as Tests

### QA Steps


#### Unable to currently test

The required back end changes are not yet finished to properly test this in the QA environment. However, once those are finished you should be able to use the same tests after completing the steps here: follow the instructions [here](https://expensify.slack.com/archives/C05EQBQG6E5/p1695182901255769), then create an report that violates one of those policies.
